### PR TITLE
Dogfood

### PR DESCRIPTION
**Pull request by bukzor**
Where I had used virtualenv/pip in the testing and dev of pip-faster, I now use pip-faster.

### CODE REVIEW

File: tests/functional/get_installed_test.py [hunk 1]
```diff
@@ -35,7 +37,7 @@ def get_installed():
 def test_pip_get_installed(tmpdir):
     tmpdir.chdir()
 
-    run('virtualenv', 'myvenv')
+    run('virtualenv', '--python', python, 'myvenv')
```
Comment (kentwills) [inline tests/functional/get_installed_test.py]: is this necessary?

File: tests/functional/pip_faster.py [hunk 2]
```diff
@@ -27,7 +29,7 @@ def it_shows_help_for_prune():
 @pytest.mark.usefixtures('pypi_server')
 def it_installs_stuff(tmpdir):
     venv = tmpdir.join('venv')
-    run('virtualenv', str(venv))
+    run('virtualenv', '--python', python, str(venv))
```
Comment (kentwills) [inline tests/functional/pip_faster.py]: ditto

File: tests/testing/__init__.py [hunk 3]
```diff
@@ -83,13 +84,28 @@ def strip_coverage_warnings(stderr):
...
 
+pip_warnings_regex = Regex(
+    r'^Coverage.py warning: (%s)\n' % '|'.join((
+        r'Module .* was never imported\.',
+        r'No data was collected\.',
+        r'Module venv_update was previously imported, but not measured\.',
+    )),
+    flags=MULTILINE,
+)
+
```
Comment (kentwills) [inline tests/testing/__init__.py]: not used

File: tests/testing/__init__.py [hunk 4]
```diff
@@ -83,13 +84,28 @@ def strip_coverage_warnings(stderr):
...
+
+def strip_pip_warnings(stderr):
+    return pip_warnings_regex.sub('', stderr)
+
```
Comment (kentwills) [inline tests/testing/__init__.py]: ditto

File: venv_update.py [hunk 5]
```diff
@@ -96,50 +81,186 @@ def info(msg):
...
+            if source_version == destination_version:
+                info('Keeping valid virtualenv from previous run.')
+                raise SystemExit(0)  # looks good! we're done here.
+
+    # TODO: say exactly *why* the venv was invalidated
+    info('Removing invalidated virtualenv.')
+    run(('rm', '-rf', venv_path))
+
+
+def ensure_virtualenv(args, return_values):   # FIXME!: too complex.
```
Comment (kentwills) [inline venv_update.py]: fixd

File: pylintrc [hunk 6]
```diff
@@ -38,7 +38,7 @@ class-rgx=%(const-rgx)s
 max-line-length=131
 
 # TODO: we'd like to hit 250 here
-max-module-lines=500
+max-module-lines=530
```
Comment (kentwills) [inline pylintrc]: drop # of lines

**Comment by kentwills** [comment]
You still have WIPs in the history

**Comment by chriskuehl** [comment]
overall looks good to me! :D

**Comment by kentwills** [comment]
History  is much cleaner, ship it!

**Pull request merged**